rudra/utils/github_utils.py

```diff
@@ -0,0 +1,70 @@
+# Copyright © 2020 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: Dharmendra G Patel <dhpatel@redhat.com>
+#
+"""Github utility providing functionality to read and write github resource."""
+from github import Github
+
+
+class GithubUtils:
+    """Class to perform action related to github like get content / raise PR etc."""
+
+    def __init__(self, github_token):
+        """Get github token to initialize the class."""
+        # Create github object using token
+        self._github = Github(github_token)
+
+    def set_repo(self, repo_path, master_ref='master'):
+        """Setup repo and master ref details."""
+        self._master_ref = master_ref
+
+        # Get repo and set to private member
+        self._repo = self._github.get_repo(repo_path)
+
+    def get_latest_commit_hash(self):
+        """Get the latest commit hash of the repo."""
+        # Read all commits for a repo.
+        commits = self._repo.get_commits()
+
+        # Read the latest commit of the repo at index '0'
+        latest_commit_hash = commits[0].sha
+
+        return latest_commit_hash
+
+    def create_branch(self, branch_name):
+        """Create a branch with latest code."""
+        # Create branch with latest commit
+        ref = self._repo.create_git_ref('refs/heads/' + branch_name,
+                                        self.get_latest_commit_hash())
+        return ref.url
+
+    def get_content(self, file_path):
+        """Read and returns content of the given file path within repo."""
+        # Read file content of a path relative to repo root directory.
+        contents = self._repo.get_contents(file_path)
```

Comment:
Which branch contents it gets?

Reply:
When we set repo, it will point to master (latest commit), so, it will read always latest master code. Even we create branch this code points branch to latest master code. I don;t see a requirement to point to any commit hash as of now. If such requirements existing in future we can extend the functionality of this class.

Reply:
It should be explicit, the reader of the api should't assume thing. Thats the reason for asking to include the branch name in the constructor.

Reply:
I guess it points to default branch of the repo, not master. default branch could be anything.better be explicit and keep it as an arg to __init__